hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -1037,7 +1059,11 @@ public void writeData(ObjectDataOutput out) throws IOException {
         out.writeObject(nearCacheConfig);
         out.writeBoolean(readBackupData);
         out.writeUTF(cacheDeserializedValues.name());
-        out.writeUTF(mergePolicy);
+        if (out.getVersion().isGreaterOrEqual(Versions.V3_10)) {
```

Comment:
`MapConfig` must implement `Versioned` to use version information

Reply:
Good catch, fixed!